### ISSUE TITLE
cursor does not properly scroll into view when scroll up in a large input cell

### ISSUE THREAD

**Issue by williamstein** [jupyter/notebook#2464]
1. create a large input cell which is well over one page in length
2. scroll down
3. hit the up arrow key to move up until going past the top of the cell list.
4. the cursor will disappear for a few up arrows until it finally scrolls back into view.

Note that scrolling down is fine; only scrolling up is a problem. 

I hit exactly the same bug in doing my rewrite of Jupyter for SageMathCloud.  It's perhaps a bug in CodeMirror, though due to how HTML/DOM works, I'm not sure if it makes sense to fix things there.  The workaround I used for SMC is https://github.com/sagemathinc/smc/commit/cd7ddc1d8ce0faf2369fe7b9fa4162b09b170283; the main point being in the file src/smc-webapp/jupyter/codemirror-editor.cjsx
Basically: watch for cursor events and compare the cell list containers offset().top with the cursor position; if one is less than the other (so cursor is not visible) use scrollTop to move the scroll position.
